File: python/api/booking.py
import requests


def all_bookings():
    return requests.get('https://restful-booker.herokuapp.com/booking')


def specific_booking(identifier, accept):
    headers = {'Accept': accept}
    return requests.get('https://restful-booker.herokuapp.com/booking/' + str(identifier), headers=headers)


def create_booking(payload, content_type):
    headers = {'Accept': 'application/json', 'Content-type': content_type}
    return requests.post('https://restful-booker.herokuapp.com/booking', payload, headers=headers)


def delete_booking(identifier, token):
    headers = {'Cookie': 'token=' + token}
    return requests.delete('https://restful-booker.herokuapp.com/booking/' + str(identifier), headers=headers)

File: js/api/booking.js
request = require('supertest-as-promised')

exports.all_bookings = function(){
  return request('https://restful-booker.herokuapp.com')
           .get('/booking')
};

exports.specific_booking = function(identifier, accept){
  return request('https://restful-booker.herokuapp.com')
          .get('/booking/' + identifier)
          .set('accept', accept)
};

exports.create_booking = function(payload, contentType){
  return request('https://restful-booker.herokuapp.com')
           .post('/booking')
           .set('accept', 'application/json')
           .set('Content-type', contentType)
           .send(payload)
};

exports.deleteBooking = function(identifier, token){
  return request('https://restful-booker.herokuapp.com')
           .delete('/booking/' + identifier)
           .set('Cookie', 'token=' + token)
}
